Libraries/LibLine/Editor.cpp

```diff
@@ -279,17 +282,20 @@ String Editor::get_line(const String& prompt)
                 m_refresh_needed = true;
                 continue;
             }
+
+            auto saved_errno = errno;
+
             perror("read failed");
-            // FIXME: exit()ing here is a bit off. Should communicate failure to caller somehow instead.
-            exit(2);
+            dbg() << "Read failed: " << strerror(saved_errno);
```

Comment:
`perror` should already print to `dbg()` log itself, so no need for `saved_errno` at all.

Unless you want to preserve `errno` for the caller; then use `ScopedValueRollback errno_restorer(errno);`

Reply:
Very well.
please tell me I didn't forget to put errno back...I did...